source/dagonBackend.d

```diff
@@ -1110,6 +1110,12 @@ final class SacScene: Scene{
 				case surrender:
 					controller.addCommand(Command!DagonBackend(renderSide));
 					break;
+				case moveUp, moveDown:
+					enforce(0,"bad hotkeys");
+					break;
+				case cameraUp, cameraDown, cameraLeft, cameraRight, cameraForward, cameraBackward:
+					enforce(0,"bad hotkeys");
+					break;
```

Comment:
Instead of duplicating the code, you can just put something like this (further up to the other movement hotkeys):

```d
case moveForward,moveBackward,turnLeft,turnRight,cameraZoomIn,cameraZoomOut:
case moveUp, moveDown:
case cameraUp, cameraDown, cameraLeft, cameraRight, cameraForward, cameraBackward:
    enforce(0,"bad hotkeys");
    break;
```

This is idiomatic in D because there is no implicit fallthrough in switch cases (you use `goto case; to explicitly go to next case).

Reply:
does this not use the fallthrough mechanic though?